src/Microsoft.Dnx.Tooling/Restore/NuGet/NuGetv3Feed.cs

```diff
@@ -213,7 +214,41 @@ public PackageInfo BuildModel(string id, string version)
 
         public async Task<Stream> OpenNuspecStreamAsync(PackageInfo package)
         {
-            return await PackageUtilities.OpenNuspecStreamFromNupkgAsync(package, OpenNupkgStreamAsync, _reports.Information);
+            for (int retry = 0; retry != 3; ++retry)
+            {
+                try
+                {
+                    var data = await _httpSource.GetAsync(
+                        package.ManifestUri,
+                        cacheKey: $"nuspec_{package.Id}.{package.Version}",
+                        cacheAgeLimit: retry == 0 ? _cacheAgeLimitNupkg : TimeSpan.Zero,
+                        ensureValidContents: stream => PackageUtilities.EnsureValidManifestContents(stream, package));
+                    return data.Stream;
+                }
+                catch (Exception ex)
+                {
+                    var isFinalAttempt = (retry == 2);
+                    var message = ex.Message;
+                    if (ex is TaskCanceledException)
+                    {
+                        message = ErrorMessageUtils.GetFriendlyTimeoutErrorMessage(ex as TaskCanceledException, isFinalAttempt, ignoreFailure: false);
+                    }
+
+                    if (isFinalAttempt)
+                    {
+                        _reports.Error.WriteLine(
+                            $"Error occurred when downloading manifest: {package.ManifestUri}{Environment.NewLine}  {message}".Red().Bold());
+                        throw;
+                    }
+                    else
+                    {
+                        _reports.Information.WriteLine(
+                            $"Warning occurred when downloading manifest: {package.ManifestUri}{Environment.NewLine}  {message}".Yellow().Bold());
```

Comment:
We already tell users that we are retrying in error message part:
![capture](https://cloud.githubusercontent.com/assets/1383883/9050513/2b2815f0-3a03-11e5-8606-c1d6acda56c6.PNG)

Moreover, we have 3 or 4 similar retry logic in DNX, we should modify all of them if we want to change this one. That's beyond the scope of this PR. Can you file an issue and label it with suggestion?